receiver/awss3receiver/notifications_test.go

```diff
@@ -0,0 +1,212 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package awss3receiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awss3receiver"
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/open-telemetry/opamp-go/client/types"
+	"github.com/open-telemetry/opamp-go/protobufs"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/pdata/plog"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/opampcustommessages"
+)
+
+type mockCustomCapabilityRegistry struct {
+	component.Component
+
+	shouldFailRegister  bool
+	shouldReturnPending bool
+
+	pendingChannel   chan struct{}
+	unregisterCalled bool
+	sentMessages     []customMessage
+}
+
+type customMessage struct {
+	messageType string
+	message     []byte
+}
+
+type hostWithCustomCapabilityRegistry struct {
+	extension *mockCustomCapabilityRegistry
+}
+
+func (h hostWithCustomCapabilityRegistry) Start(context.Context, component.Host) error {
+	panic("unsupported")
+}
+
+func (h hostWithCustomCapabilityRegistry) Shutdown(context.Context) error {
+	panic("unsupported")
+}
+
+func (h hostWithCustomCapabilityRegistry) GetFactory(_ component.Kind, _ component.Type) component.Factory {
+	panic("unsupported")
+}
+
+func (h hostWithCustomCapabilityRegistry) GetExtensions() map[component.ID]component.Component {
+	return map[component.ID]component.Component{
+		component.MustNewID("foo"): h.extension,
+	}
+}
+
+func (h hostWithCustomCapabilityRegistry) GetExporters() map[component.DataType]map[component.ID]component.Component {
+	panic("unsupported")
+}
+
+func (m *mockCustomCapabilityRegistry) Register(_ string, _ ...opampcustommessages.CustomCapabilityRegisterOption) (handler opampcustommessages.CustomCapabilityHandler, err error) {
+	if m.shouldFailRegister {
+		return nil, fmt.Errorf("register failed")
+	}
+	return m, nil
+}
+
+func (m *mockCustomCapabilityRegistry) Message() <-chan *protobufs.CustomMessage {
+	panic("unsupported")
+}
+
+func (m *mockCustomCapabilityRegistry) SendMessage(messageType string, message []byte) (messageSendingChannel chan struct{}, err error) {
+	if m.unregisterCalled {
+		return nil, fmt.Errorf("unregister called")
+	}
+	m.sentMessages = append(m.sentMessages, customMessage{messageType: messageType, message: message})
+	if m.shouldReturnPending {
+		return m.pendingChannel, types.ErrCustomMessagePending
+	}
```

Comment:
```suggestion
	if m.shouldReturnPending {
		return m.pendingChannel, types.ErrCustomMessagePending
	}
	m.sentMessages = append(m.sentMessages, customMessage{messageType: messageType, message: message})
```
I think we need to not record a message as sent if one is pending. The [client implementation](https://github.com/open-telemetry/opamp-go/blob/main/client/internal/clientcommon.go#L403) doesn't allow sending messages no messages are currently queued.